local/network.go

```diff
@@ -66,8 +66,18 @@ type localNetwork struct {
 	newAPIClientF api.NewAPIClientF
 	// Used to create new node processes
 	nodeProcessCreator NodeProcessCreator
-	// Closed when network is done shutting down
-	closedOnStopCh chan struct{}
+	stopOnce           sync.Once
+	// Cancelled when Stop begins.
+	onStopCtx context.Context
```

Comment:
Is using a context more idiomatic than using a channel? I prefer reducing the number of used fields for a given solution. It seems the context is not wrapped by any other context, so a channel can be used instead. Whould like to know you opinion.

Reply:
changed to `onStopCh`